### PR TITLE
More hook test cleanup

### PR DESCRIPTION
**Pull request by mpeyper**
So much unnecessary `act`ing in these tests.... and tests not actually testing anything

### CODE REVIEW

File: src/__tests__/shared/components/network-status/NetworkStatusMessage.test.tsx [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
-}
+describe('NetworkStatusMessage', () => {
+  const useNetworkStatusMock = (useNetworkStatus as unknown) as jest.MockInstance<
+    ReturnType<typeof useNetworkStatus>,
+    any
+  >
 
-const { result } = renderHook(() => useTranslation() as any)
-result.current.t = (key: keyof typeof englishTranslationsMock) => englishTranslationsMock[key]
-const { t } = result.current
```
Comment (mpeyper) [inline src/__tests__/shared/components/network-status/NetworkStatusMessage.test.tsx:22]: This whole bit was really strange.   It took me ages to realise that `useTranslation` is being mocked and not actually a hook any more so actually rendering it it pointless and confusing.

File: src/__tests__/shared/hooks/useTranslator.test.ts [hunk 2]
```diff
@@ -11,12 +11,11 @@ describe('useTranslator', () => {
   })
 
   it('should return useTranslation hook result if input value is NOT undefined', () => {
-    const { result: useTranslatorResult } = renderHook(() => useTranslator())
-    const { result: useTranslationResult } = renderHook(() => useTranslation())
```
Comment (mpeyper) [inline src/__tests__/shared/hooks/useTranslator.test.ts:15]: same as above, this is actually being mocked so no reason to render it.

File: src/__tests__/test-utils/use-mutation.util.ts [hunk 3]
```diff
@@ -1,19 +1,16 @@
...
-export default async function executeMutation(callback: any, ...input: any[]) {
-  let mutateToTest: any
-  await act(async () => {
-    const renderHookResult = renderHook(callback)
-    const { result, waitForNextUpdate } = renderHookResult
-    await waitForNextUpdate()
-    const [mutate] = result.current as any
-    mutateToTest = mutate
-  })
+export default async function executeMutation<TResult>(
```
Comment (mpeyper) [inline src/__tests__/test-utils/use-mutation.util.ts:4]: Making this generic is what caused most of the changes in this review.  The cleanup here is actually a drop in replacement for the old version.

File: src/__tests__/test-utils/use-mutation.util.ts [hunk 4]
```diff
@@ -1,19 +1,16 @@
 import { act, renderHook } from '@testing-library/react-hooks'
+import { MutateFunction } from 'react-query'
 
-export default async function executeMutation(callback: any, ...input: any[]) {
-  let mutateToTest: any
-  await act(async () => {
-    const renderHookResult = renderHook(callback)
-    const { result, waitForNextUpdate } = renderHookResult
-    await waitForNextUpdate()
```
Comment (mpeyper) [inline src/__tests__/test-utils/use-mutation.util.ts:8]: I'm still unsure how this was working before.  Must be some shenanigans from being wrapped in `act` but there was actually no reason to `waitForNextUpdate` here.

**Comment by JacobMGEvans** [comment]
Something satisfying having the Hooks Library author helping fix Hooks tests lmao

**Pull request merged**